docs/reference/elasticsearch/mapping-reference/text.md

```diff
@@ -341,3 +342,82 @@ The following mapping parameters are accepted:
 :   Metadata about the field.
 
 
+## Pattern text field type [pattern-text-field-type]
+```{applies_to}
+serverless: preview
+stack: preview 9.2
+```
+
+::::{warning}
+This functionality is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features.
+::::
+
+A variant of [`text`](#text-field-type) with improved space efficiency for log data.
+Internally, it decomposes values into static parts that are likely to be shared among many values, and dynamic parts that tend to vary.
+The static parts usually come from the explanatory text of a log message, while the dynamic parts are the variables that were interpolated into the logs.
+This decomposition allows for improved compression on log-like data.
+
+We call the static portion of the value the `template`.
+Although the template cannot be accessed directly, a separate field called `<field_name>.template_id` is accessible.
+This field is a hash of the template and can be used to group similar values.
+As this feature is in technical preview, the internal structure of the template is subject to change.
```

Comment:
```suggestion
```

This is implied in the technical preview warning, as the feature moves out of preview we can use additional applies_to tags in future versions to demarcate changes.